lib/src/otx/data/factory.py

```
@@ -71,12 +76,19 @@ def create(
             return OTXAnomalyDataset(task_type=task, **common_kwargs)
 
         if task == OTXTaskType.MULTI_CLASS_CLS:
-            from .dataset.classification import OTXMulticlassClsDataset
-
+            from .dataset.classification_new import ClassificationSample, OTXMulticlassClsDataset
+
+            if isinstance(dm_subset, DmDataset):
+                categories = cls._get_label_categories(dm_subset, data_format)
+                dataset = DatasetNew(ClassificationSample, categories={"label": categories})
+                for item in dm_subset:
+                    if len(item.media.data.shape) == 3:  # TODO: Account for grayscale images
+                        dataset.append(ClassificationSample.from_dm_item(item))
+                common_kwargs["dm_subset"] = dataset
```

Could you add an `else` here? If it’s not supported, we can raise a RuntimeError.

---

This might raise some problems if used with Geti. In my dataset of car images, I (unknowingly) had a grayscale image. This image has been exported in arrow format and reloaded with datumaro, but it only has 2 dimensions. Raising an error means we completely fail training. I could convert those image to be in RGB format for now or log a warning, but I don't think we should raise an error as of now.

---

Sorry for the confusion, I mean for the outer if: `if isinstance(dm_subset, DmDataset):`.